pkgs/by-name/me/med/package.nix

```diff
@@ -0,0 +1,30 @@
+{ stdenv, lib, qt6, fetchFromGitHub, cmake, pkg-config, jsoncpp, readline }:
+
+stdenv.mkDerivation rec {
+  pname = "med";
+  version = "3.10.1";
+
+  src = fetchFromGitHub {
+    owner = "allencch";
+    repo = "med";
+    rev = version;
+    sha256 = "sha256-m2lVRSNaklB0Xfqgtyc0lNWXfTD8wTWsE06eGv4FOBE=";
+  };
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook cmake pkg-config ];
+  buildInputs = [ qt6.qtbase qt6.qttools qt6.qtwayland jsoncpp readline ];
+
+  prePatch = ''
```

Comment:
```suggestion
  postPatch = ''
```

Reply:
I changed it to `postPatch`. 

Can you explain why the `postPatch` hook is preferred over `prePatch`?

Reply:
Let's say I want to apply some patch from upstream in my configuration:
```nix
nixpkgs.overlays = [
  (self: super: {
    med = super.med.overrideAttrs ({
      patches = [
        (super.fetchpatch {
          url = "https://raw.githubusercontent.com/allencch/med/...patch";
          hash = "...";
        })
      ];
    });
  })
];
```
Conflict can happen when the same file modified by `prePatch` and by the patch. Or the patch can introduce extra `/opt/med` that should be replaced.